asmcomp/power/emit.mlp

```diff
@@ -416,15 +422,26 @@ module BR = Branch_relaxation.Make (struct
         else 5
     | Lop(Istackoffset _) -> 1
     | Lop(Iload {memory_chunk; addressing_mode; _ }) ->
+      let loadinstr =
+        match memory_chunk with
+        | Thirtytwo_signed -> "lwa"
+        | Word_int | Word_val -> "ld"
+        | _ -> "neither_lwa_nor_ld"
+      in
       if memory_chunk = Byte_signed
-      then load_store_size addressing_mode + 1
-      else load_store_size addressing_mode
+      then load_store_size loadinstr addressing_mode + 1
+      else load_store_size loadinstr addressing_mode
     | Lop(Istore(chunk, addr, assignment)) ->
+      let loadinstr =
+        match chunk with
+        | Word_int | Word_val -> "std"
+        | _ -> "not_std"
+        in
         (match chunk with
          | Single -> 1
          | Word_int | Word_val when assignment -> 1
          | _ -> 0)
```

Comment:
In the spirit of sharing more code between the emitter functions and this one, I think that it would be nice to introduce a `store_needs_lwsync` helper that would be used here and there, so the code here would become `(if chunk = Single then 1 else 0) + (if store_needs_lwsync then 1 else 0) + ....`.

Reply:
Done.